### PR TITLE
2440/uni merge

### LINKED ISSUE #2440: Merge Uniswap
You know the drill...

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

(recreating from https://github.com/gnosis/cowswap/pull/2529)

Closes #2440 

Merge with latest from original UNI code

PR deployment URL https://pr2536--gpswapui.review.gnosisdev.com/

## TO DOs

List of things that I know we should look at. There might be more...

- [ ] Identicon looking weird, need to be revisited
- [ ] Network switcher: Styles
- [x] Network switcher: On hover effect
- [x] Network switcher: No longer disabled for unsupported wallets
- [ ] Widget command not starting - probably need to adjust the paths in the /lib/ files
- [x] connectors file might need to be updated
- [x] useCurrency file might need to be updated
- [ ] ErrorBoundary HTML changes we might want to apply
- [x] AdvancedSwapDropdow file has been removed from original but we still have the mod
- [ ] ListUpdatePopup file has been removed from original but we still have the mod
- [x] UrlWarning file has been removed from original but we still have the mod -- we do want to keep it. [Older merge](https://github.com/gnosis/cowswap/blob/7900f01ae5b93d56bd832339c830d6636f61f28f/src/state/user/actions.ts#L37)
- [x] Native currency on activity details has no icon because it's not tagged as native currency (eeee... address)
- [x] Probably will need to fix integration tests...
- [x] Locale/plurals are all screwed
- [ ] Added app Suspense Loader from Uni. We might want to customize it
- [x] There's a component unreacheable on SwapMod
- [ ] There's a new component that could be used: AccountDetails/index:StatusIcon
- [x] Need to re-do useChangeNetworks based on changes from NetworkSelectorMod
- [ ] Need to understand and fix whatever `rules-of-hooks` is broken on SwapMod. Temporarily disabled the rule
- [x] WETH/wxDAI -> USDC pair is pre-selected, but:
    - [x] Price for USDC is not loaded
    - [x] USDC is not selected on gchain
- [x] ESlist on CI
- [x] Loading balances via Wallet Connect is broken for some mobile wallets (MM, imToken, Token Pocket, Pillar, Argent). Worked flawlessly with Coinbase mobile wallet though
- [ ] Wallet connect wallet images no longer shown

# To Test

1. TEST EVERYTHING!!

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -72,3 +72,6 @@ REACT_APP_PATH_REGEX_ENS="/ipfs"
 
 # Enables mock mode (default = true)
 REACT_APP_MOCK=true
+
+# Locales
+REACT_APP_LOCALES="locales"
```
Comment (W3stside) [inline .env:77]: this is already concerning, why is .env no longer ignored?

File: .eslintrc.json [hunk 2]
```diff
@@ -8,21 +8,50 @@
...
     "plugin:react-hooks/recommended",
     "prettier/@typescript-eslint",
+    "plugin:cypress/recommended",
     "plugin:prettier/recommended",
     "plugin:jsx-a11y/recommended"
   ],
+  "plugins": ["simple-import-sort", "unused-imports"],
   "rules": {
+    "unused-imports/no-unused-imports": "error",
+    //    "simple-import-sort/imports": "error",
```
Comment (W3stside) [inline .eslintrc.json:53]: they removed this or we did? tbh i always liked this rule but it can get noisy

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -145,10 +145,15 @@ jobs:
...
-          eslint: true
-          prettier: true
           github_token: ${{ secrets.github_token }}
+          eslint: true
+          eslint_args: "-c .eslintrc.json ."
           auto_fix: ${{ github.event_name == 'pull_request' }}
+          prettier: true
+
+      - name: Run eslint
+        if: ${{ github.event_name == 'pull_request' && github.event.pull_request.head.repo.owner.login != github.repository_owner }}
```
Comment (W3stside) [inline .github/workflows/ci.yml:155]: weird rule, don't run eslint if the owner is opening a PR?

File: cosmos.config.json [hunk 4]
```diff
@@ -1,7 +1,8 @@
 {
-  "staticPath": "public",
   "watchDirs": ["src"],
   "webpack": {
-    "configPath": "./cosmos.webpack.config"
-  }
+    "configPath": "react-scripts/config/webpack.config",
+    "overridePath": "cosmos.override.js"
+  },
+  "port": 5001
```
Comment (W3stside) [inline cosmos.config.json:7]: so cosmos stuff willa lso be run on yarn start? seems bulkier. but ill give it a go later. just a note

File: cypress-custom/support/commands.d.ts [hunk 5]
```diff
@@ -1,4 +1,4 @@
-/// <reference types="cypress" />
+/// <reference types="Cypress" />
```
Comment (W3stside) [inline cypress-custom/support/commands.d.ts:1]: ?? 🤔

File: package.json [hunk 6]
```diff
@@ -67,99 +75,55 @@
...
-    "ua-parser-js": "^0.7.28",
-    "use-count-up": "^2.2.5",
-    "wcag-contrast": "^3.0.0",
-    "web-vitals": "^2.1.0",
-    "web3": "^1.7.0",
-    "web3-providers-http": "^1.7.0",
-    "web3-providers-ws": "^1.7.0",
+    "typescript": "^4.4.3",
+    "web3-react-abstract-connector": "npm:@web3-react/abstract-connector@^6.0.7",
+    "web3-react-core": "npm:@web3-react/core@^6.0.9",
```
Comment (W3stside) [inline package.json:113]: whta's this npm prepending going on?

File: package.json [hunk 7]
```diff
@@ -67,99 +75,55 @@
...
+    "web3-react-fortmatic-connector": "npm:@web3-react/fortmatic-connector@^6.0.9",
+    "web3-react-injected-connector": "npm:@web3-react/injected-connector@^6.0.7",
+    "web3-react-portis-connector": "npm:@web3-react/portis-connector@^6.0.9",
+    "web3-react-types": "npm:@web3-react/types@^6.0.7",
+    "web3-react-walletconnect-connector": "npm:@web3-react/walletconnect-connector@^7.0.2-alpha.0",
```
Comment (W3stside) [inline package.json:118]: beta connector in use, differs quite alot from v6

File: package.json [hunk 8]
```diff
@@ -219,24 +188,144 @@
...
+    "@uniswap/v3-core": "1.0.0",
+    "@uniswap/v3-periphery": "^1.1.1",
+    "@uniswap/v3-sdk": "^3.8.2",
     "@walletconnect/web3-provider": "^1.6.6",
+    "@web3-react/core": "8.0.14-beta.0",
+    "@web3-react/eip1193": "8.0.9-beta.0",
+    "@web3-react/empty": "8.0.7-beta.0",
+    "@web3-react/fortmatic-connector": "^6.1.6",
+    "@web3-react/injected-connector": "^6.0.7",
+    "@web3-react/metamask": "8.0.10-beta.0",
```
Comment (W3stside) [inline package.json:251]: sorry made a mistake above, these are all the latest connectors

File: src/components/AddressInputPanel/index.tsx [hunk 9]
```diff
@@ -1,11 +1,12 @@
+import { Trans } from '@lingui/macro'
 // eslint-disable-next-line no-restricted-imports
-import { t, Trans } from '@lingui/macro'
+import { t } from '@lingui/macro'
```
Comment (W3stside) [inline src/components/AddressInputPanel/index.tsx:3]: should combine these, is this a mistake?

**Comment by elena-zh** [comment]
Hey @alfetopito , great job!
I have done a round of PR testing and reported related issues: 
- https://github.com/gnosis/cowswap/issues/2626
- https://github.com/gnosis/cowswap/issues/2627
- https://github.com/gnosis/cowswap/issues/2629
- https://github.com/gnosis/cowswap/issues/2630
- https://github.com/gnosis/cowswap/issues/2640
- https://github.com/gnosis/cowswap/issues/2641
- https://github.com/gnosis/cowswap/issues/2645

Besides these issues, I'd like to leave here some notes: 
1. I have not fully tested Profile page as it should be updated according to the latest changes
2. I have some doubts in displaying WETH-USDC token pair b default. Seems like the app forces me immediately to trade with these tokens. Besides. when switch to Rinkeby, it shows an incorrect price.. So I'd leave TO field empty as we do currently in Prod. However, I do not insist on it 
incorrect USDC price
3. I see a lot of errors in the Console. Seems like it is totally red
failed requests

Thanks!

**Pull request closed without merging**